msticpy/data/drivers/sumologic_driver.py

```diff
@@ -425,9 +527,31 @@ def query(
         else:
             dataframe_res = pd.DataFrame(results)
 
+        if limit is not None and dataframe_res.shape[0] > limit:
+            dataframe_res = dataframe_res.head(limit)
+
         for col in dataframe_res.columns:
-            if col in ("map._count", "map._timeslice"):
-                dataframe_res[col] = pd.to_numeric(dataframe_res[col])
+            try:
+                if (
+                    col
+                    in ["map._count", "map._collectorid", "map._messageid", "map._size"]
+                    + numeric_columns
+                ):
+                    dataframe_res[col] = pd.to_numeric(dataframe_res[col])
```

Comment:
Apparently dynamically adding pandas columns like this can result in DF fragmentation, which results in warnings from pandas (and presumably poorer perf).
One option would be to return a df.copy() at the end of the function.

Reply:
changed